### PR TITLE
eslint-3: Added ESLint plugins

### PR DESCRIPTION
**Pull request by kunagpal**
* [JSDoc](https://github.com/gajus/eslint-plugin-jsdoc)
* [Lodash](https://github.com/wix/eslint-plugin-lodash)
* [Mocha](https://github.com/lo1tuma/eslint-plugin-mocha)
* [MongoDB](https://github.com/nfroidure/eslint-plugin-mongodb)
* [Node Security Project](https://github.com/nodesecurity/eslint-plugin-security)

### CODE REVIEW

**Comment by dblandin** [comment]
Hey @kunagpal,

Thanks for the PR! These plugins look like great additions to me.

Would you mind adding the appropriate category mappings for non-style rules in [lib/checks.js](https://github.com/codeclimate/codeclimate-eslint/blob/channel/eslint-3/lib/checks.js#L1)?

**Comment by kunagpal** [comment]
@dblandin Sure thing, I'll update this pull request and get back to you.

**Comment by kunagpal** [comment]
@dblandin The [lib/checks.js](https://github.com/kunagpal/codeclimate-eslint/blob/channel/eslint-3-plugins/lib/checks.js) file looks quite bloated now, with a majority of the file lines (208 / 267) being dominated by the `checkCategoryMapping` object. This should probably be shifted to an independent JSON within `lib`.

**Comment by dblandin** [comment]
> @dblandin lib/checks.js looks quite bloated now, with a majority of the file lines (208 / 267) being dominated by the checkCategoryMapping object. This should probably be shifted to an independent JSON within lib.

I agree! Would be happy to see that file broken up either in this PR or a separate preceding refactor PR.

**Comment by kunagpal** [comment]
@dblandin Cool, will segregate in the next commit.

File: lib/check_category_mapping.json [hunk 1]
```diff
@@ -0,0 +1,208 @@
...
+  "jsdoc/require-param-type": "Clarity",
+  "jsdoc/require-returns-description": "Clarity",
+  "jsdoc/require-returns-type": "Clarity",
+
+  "lodash/callback-binding": "Compatibility",
+  "lodash/collection-method-value": "Bug Risk",
+  "lodash/collection-return": "Bug Risk",
+  "lodash/no-double-unwrap": "Bug Risk",
+  "lodash/no-extra-args": "Clarity",
+  "lodash/unwrap": "Bug Risk",
```
Comment (dblandin) [inline lib/check_category_mapping.json]: This looks great. I like the plugin-specific sections. Thanks!

Would you mind running these sections through an alphabetical sort? Looks like they're a bit out of order.

**Comment by dblandin** [comment]
One minor comment. Once addressed, this LGTM!

File: lib/check_category_mapping.json (hunk 1 above)
Comment (kunagpal) [inline lib/check_category_mapping.json]: Sure thing.

**Comment by kunagpal** [comment]
@dblandin Out of curiosity, when will this change become available for use?

**Comment by kunagpal** [comment]
Thanks @maxjacobson!

**Comment by maxjacobson** [comment]
Thank you for the contribution :)

**Pull request merged**